src/Client/UnixStream.hpp

```diff
@@ -167,10 +168,8 @@ UnixStream::connect(const ConnectOptions &opts_arg)
 			      addr_info.last_error());
 	int socket_errno = 0, connect_errno = 0;
 	for (auto &inf: addr_info) {
-		fd = ::socket(inf.ai_family,
-			      inf.ai_socktype | SOCK_NONBLOCK | SOCK_CLOEXEC,
-			      inf.ai_protocol);
-		if (fd < 0) {
+		fd = ::socket(inf.ai_family, inf.ai_socktype, inf.ai_protocol);
+		if (fd < 0 || fcntl(fd, F_SETFD, O_NONBLOCK | O_CLOEXEC) < 0) {
```

Comment:
It seems, `O_CLOEXEC` should be replaced with `FD_CLOEXEC` (see https://github.com/tarantool/tarantool/commit/e9abaef0b7e5758371ed023af9cda99a48330d15) and, I guess, `O_NONBLOCK` should be set using `F_SETFL` instead of `F_SETFD` (according to Linux's man page `fcntl(2)`).

Reply:
Fixed both comments, thanks.